fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/BestMvSelector.java

```diff
@@ -152,15 +178,38 @@ private void calculateStatistics(OptExpression expr, OptimizerContext context) {
         expr.setStatistics(expressionContext.getStatistics());
     }
 
+    private int calcSortScore(MaterializedView mv, Set<String> equivalenceColumns, Set<String> nonEquivalenceColumns) {
+        List<Column> keyColumns = mv.getKeyColumnsByIndexId(mv.getBaseIndexId());
+        int score = 0;
+        for (Column col : keyColumns) {
+            String columName = col.getName().toLowerCase();
+            if (equivalenceColumns.contains(columName)) {
+                score++;
+            } else if (nonEquivalenceColumns.contains(columName)) {
+                // UnEquivalence predicate's columns can only match first columns in rollup.
+                score++;
```

Comment:
why are equal and nonequal's socres the same?

Reply:
they are not the same. only the first nonequal can be counted.